pkg/httpsig/algorithm.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package httpsig
+
+//nolint:lll
+//go:generate mockgen -destination gomocks_test.go -package httpsig_test -source=algorithm.go -mock_names keyResolver=MockKeyResolver
+
+import (
+	"crypto/ed25519"
+	"errors"
+	"fmt"
+
+	ariesverifier "github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+	httpsig "github.com/igor-pavlenko/httpsignatures-go"
+)
+
+const aceHTTPSigAlgorithm = "Ed25519"
+
+// ErrInvalidSignature indicates that the signature is not valid for the given data.
+var ErrInvalidSignature = errors.New("invalid HTTP signature")
+
+type keyResolver interface {
+	// Resolve returns the public key bytes and the type of public key for the given key ID.
+	Resolve(keyID string) (*ariesverifier.PublicKey, error)
+}
+
+// SignatureHashAlgorithm is a custom httpsignatures.SignatureHashAlgorithm that uses ed25519 key to sign HTTP requests.
+type SignatureHashAlgorithm struct {
+	pubKeyResolver keyResolver
+	privateKey     ed25519.PrivateKey
```

Comment:
You have a handle to private key material here, you may want to use Signer interface instead. This can be addressed in later PRs if it is an issue (depends on use cases).